### PR TITLE
RE2022-89: Add a heartbeat to match processes

### PR DESCRIPTION
**Pull request by MrCreosote**
Tested by adding an asyncio.sleep(100) statement to the match process and watching the match record in arango

Next:
* When a standard match or data product match is pulled from the DB, check the last heart beat and restart the process if it's beyond the threshold

### CODE REVIEW

File: src/service/processing.py [hunk 1]
```diff
@@ -45,3 +52,48 @@ def _run_process(
     args: list[Any]
 ):
     asyncio.run(function(data_id, pstorage, args))
+
+
+class Heartbeat:
```
Comment (Tianhao-Gu) [inline src/service/processing.py:57]: Do you think in the future maybe we need something like a `status` method to check the current status of the heartbeat, whether it's running or stopped

File: src/service/processing.py [hunk 2]
```diff
@@ -45,3 +52,48 @@ def _run_process(
...
+        self._schd.start()
+
+    async def _heartbeat(self):
+        await self._hbf(now_epoch_millis())
+
+    def stop(self):
+        """
+        Stop the heartbeat.
+        """
+        if self._job_id:
```
Comment (Tianhao-Gu) [inline src/service/processing.py:96]: Do we need to handle the case where the _job_id is None or the job has already been removed?

**Review by Tianhao-Gu — APPROVED** [review]
LGTM.

File: src/service/processing.py (hunk 2 above)
Comment (MrCreosote) [reply src/service/processing.py:96]: I'm not sure I understand - this does handle that case by not doing anything. I don't think there's anything to do - what were you thinking?

File: src/service/processing.py (hunk 1 above)
Comment (MrCreosote) [reply src/service/processing.py:57]: Based on the current usage there's no need for it, but it'd be easy to add if a use case turns up. Right now it's just start it, forget it, and then stop it when your process is complete

Comment (Tianhao-Gu) [reply src/service/processing.py:57]: 👍

File: src/service/processing.py (hunk 2 above)
Comment (Tianhao-Gu) [reply src/service/processing.py:96]: yea. I agree. I was just thinking something like a print statement in an `else` clause. It probably won't make any difference I agree.

**Pull request merged**